[mergebot] make merge on green default

zengk95 opened the pull request:
After discussing with the team, we wanted to make the wait for required status checks to be the default behavior instead of the early quit.

davidberard98 committed:
[JIT] enable autocasting + freezing test

Test was marked as `skip` due ot a memory leak. Turns out the memory leak is expected - it can be fixed by clearing the compilation unit (with `torch.jit._state._python_cu.drop_all_functions()` at the end of the test function) or by disabling the leak detector on this test.

Fixes #77618

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78566

Approved by: https://github.com/eellison

mrshenli committed:
Avoid CPU Sync in SyncBatchNorm When Capturing CUDA Graphs

We recently updated `SyncBatchNorm` to support empty input batches.
The new code removes stats from ranks with empty inputs. However,
this change breaks CUDA graph capture as it forces CPU sync. This
commit uses `is_current_stream_capturing()` to guard the new code
path, and only run the new code when not capturing CUA Graphs. To
support empty inputs with CUDA graph capturing, we might need to
update CUDA kernels for `batch_norm_backward_elemt` and
`batch_norm_gather_stats_with_counts`. See #78656.

Fixes #78549

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78666

Approved by: https://github.com/albanD

eellison committed:
Add Caching of Conversion to Fake/Meta tensors in FakeTensorMode

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78090

Approved by: https://github.com/ezyang

zengk95 committed:
make merge on green default

zengk95 committed:
make merge on green default

anjali411 committed:
Autogen Tags enum, and allow specifying tags while defining an op

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77313

Approved by: https://github.com/ezyang, https://github.com/albanD

Gamrix committed:
Updating torch.log example

Fixes issue  #78301

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78776

Approved by: https://github.com/ngimel

kit1980 committed:
Migrate off Xenial gcc5.4 for trunk jobs (#78734)

Fixes #78732
Pull Request resolved: https://github.com/pytorch/pytorch/pull/78734
Approved by: https://github.com/seemethere

Commit:
[FSDP] Allow different `optim_input` orders across ranks

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78599

Approved by: https://github.com/rohan-varma

Commit:
[FSDP][Docs] Fix typo in `full_optim_state_dict()`

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78784

Approved by: https://github.com/rohan-varma

pytorchmergebot committed:
Revert "Reenable assert after test update"

This reverts commit b0814b63df40132952f212129e12f4c8bab6bf94.

Reverted https://github.com/pytorch/pytorch/pull/78658 on behalf of https://github.com/malfet due to test_ops crashes with SIGIOT on both PR and trunk CI, see https://hud.pytorch.org/pytorch/pytorch/commit/b0814b63df40132952f212129e12f4c8bab6bf94

eqy committed:
[cuDNN][TF32] Threshold adjustments for TF32 on `>=sm80` (#78437)

CC @ptrblck @mcarilli

Change to transformer multilayer test can potentially be swapped in favor of an rtol change? (see also: #75612).
Pull Request resolved: https://github.com/pytorch/pytorch/pull/78437
Approved by: https://github.com/ngimel

eqy committed:
[TF32] Fix typo in tf32 wrapper function (#78438)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78438
Approved by: https://github.com/ngimel

jerryzh168 committed:
[quant] follow up fixes for prepare_fx/prepare_qat_fx calls in classyvision (#105) (#78660)

Summary:
X-link: https://github.com/fairinternal/ClassyVision/pull/105

As follow up for https://github.com/pytorch/pytorch/pull/76496, we fixes the TODOs in quantization tests
by providing correct example_inputs in the tests

Test Plan:
classyvision sandcastle and ossci

**Static Docs Preview: classyvision**
|[Full Site](https://our.intern.facebook.com/intern/staticdocs/eph/D36818665/V1/classyvision/)|

|**Modified Pages**|

Differential Revision: D36818665

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78660
Approved by: https://github.com/vkuzo

Chillee committed:
Ported t decomp to become a ref (#78686)

Also added an error input for `t`
Pull Request resolved: https://github.com/pytorch/pytorch/pull/78686
Approved by: https://github.com/mruberry

justinchuby committed:
[ONNX] Variable length argument support for quantized_args (#78775)

Add support for decorating functions with variable length arguments in `quantized_args`. This is needed to decorate functions like `symbolic_fn` in `_interpolate_helper` which takes `*args`.

Previously it is not possible to decorate functions like it. Now we can do

```python
@quantized_args(True)
def symbolic_fn(g, input, output_size, *args):
    ...
```

and the rest of the params are defaulted to non-quantized.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/78775
Approved by: https://github.com/garymm

pytorchmergebot committed:
Revert "Autogen Tags enum, and allow specifying tags while defining an op"

This reverts commit 9476a78f3754aa122323b431c59360b254559d16.

Reverted https://github.com/pytorch/pytorch/pull/77313 on behalf of https://github.com/malfet due to Broke OSS buck builds, see https://hud.pytorch.org/pytorch/pytorch/commit/9476a78f3754aa122323b431c59360b254559d16

kit1980 committed:
Add linux-focal-py3.7-gcc7 to merge_rules.json (#78785)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78785
Approved by: https://github.com/malfet

kit1980 committed:
Remove non-existing code_template.py glob (#78773)

Test Plan: No-op, rely on CI

Reviewed By: dagitses

Differential Revision: D36848770

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78773
Approved by: https://github.com/linbinyu, https://github.com/seemethere

LinjianMa committed:
[FSDP] Return original module when fsdp wrapped model call .module (#78671)

Fixes #78607

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78671
Approved by: https://github.com/awgu, https://github.com/rohan-varma

eqy committed:
[cuDNN] [cuDNN v8 API] Support cuDNN Errata Filter (#73934)

Not originally mentioned in the tracking issue #58414, but is a nice-to-have feature. In summary, the errata filter allows known problematic kernels to be skipped instead of irrecoverably crashing a CUDA context (e.g., via an illegal memory access) via a JSON file supplied at run time. cuDNN frontend description: https://github.com/NVIDIA/cudnn-frontend#errata-filter

Sample errata filter JSON:
```
{
  "version" : 1,
  "rules" : [
    {
      "rule_id" : "avoid_bad_bwd_data",
      "operation" : "ConvBwdData",
      "engine" : 12,
      "cudnn_version_start" : 8000,
      "cudnn_version_end" : 9000
    }
  ]
}
```
CC @ngimel @zasdfgbnm @ptrblck

Pull Request resolved: https://github.com/pytorch/pytorch/pull/73934
Approved by: https://github.com/ngimel

linbinyu committed:
rename BUILD.buck to BUCK.oss (#78792)

rename BUILD.buck to BUCK.oss to better reflect that it's the OSS version of BUCK build, not the one shared with Bazel
Pull Request resolved: https://github.com/pytorch/pytorch/pull/78792
Approved by: https://github.com/kit1980

janeyx99 commented on .github/scripts/trymerge.py:
```diff
@@ -913,16 +912,11 @@ def handle_exception(e: Exception, msg: str = "Merge failed") -> None:
-        except Exception as e:
-            handle_exception(e)
+    try:
+        merge_on_green(args.pr_num, repo, dry_run=args.dry_run)
```
will we also rename this to normal merge?

YifanShenSZ committed:
to_padded_tensor doc v0 (#78657)

Fixes #76846

Pull Request resolved: https://github.com/pytorch/pytorch/pull/78657
Approved by: https://github.com/jbschlosser

zengk95 committed:
Merge branch 'default-on-green' of https://github.com/pytorch/pytorch into default-on-green

zengk95 committed:
fixed a name
